Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.m

```diff
@@ -47,7 +47,7 @@ - (id)initWithItem:(NSXMLElement *)item
 		NSString *jidStr = [item attributeStringValueForName:@"jid"];
 		jid = [[XMPPJID jidWithString:jidStr] bareJID];
 		
-		itemAttributes = [item attributesAsDictionary];
+		itemAttributes = [[item attributesAsDictionary] mutableCopy];
```

Comment:
Is this change necessary?

Reply:
@t-unit: Have you looked the @chrisballinger comment?

Reply:
@chrisballinger KissXML does expose this method with an immutable return value – see https://github.com/robbiehanson/KissXML/blob/6db0bcbb5a15687846f0c1a0db8d0fa5a2d3a7dc/KissXML/Additions/DDXMLElementAdditions.h#L23

Reply:
@t-unit Yes exactly, so wouldn't we want to revert the `mutableCopy` changes on this line?

Reply:
@chrisballinger Don't quite understand. The [`itemAttributes`](https://github.com/robbiehanson/XMPPFramework/blob/c20cafccd422c513f78a108aa20fbce9b79ab7e8/Extensions/Roster/MemoryStorage/XMPPUserMemoryStorageObject.h#L17) property is defined a mutable – so it should not get an immutable dictionary assigned to it.

Reply:
@t-unit Ahhhh now I get it, I wasn't looking closely enough. The reason for the change is that bundled internal version of `attributesAsDictionary` returns a mutable value, and the very similar KissXML version returns it as immutable. Looks good to me!